tilewe/src/ctilewemodule.c

```diff
@@ -225,6 +225,89 @@ static PyObject* Board_NumPlayerPcs(BoardObject* self, PyObject* args, PyObject*
     return NULL; 
 }
 
+static PyObject* Board_PlayerOpenCorners(BoardObject* self, PyObject* args, PyObject* kwds) 
+{
+    static const char* kwlist[] = 
+    {
+        "for_player", 
+        NULL
+    };
+
+    int player = Tw_Color_None; 
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "|i", kwlist, &player)) 
+    {
+        return NULL; 
+    }
+
+    if (player == Tw_Color_None) 
+    {
+        player = self->Board.CurTurn; 
+    }
+
+    if (player >= 0 && player < self->Board.NumPlayers) 
+    {
+        // count the open corners
+        Py_ssize_t openCorners = 0;
+        Tw_TileSet_FOR_EACH(self->Board.Players[player].OpenCorners.Keys, tile, 
+        {
+            openCorners++;
+        });
+
+        // build Python list of the open corner tiles
+        PyObject* list = PyList_New(openCorners); 
+        Py_ssize_t cornerIndex = 0;
+        Tw_TileSet_FOR_EACH(self->Board.Players[player].OpenCorners.Keys, tile, 
+        {
+            PyList_SetItem(
+                list, 
+                cornerIndex, 
+                PyLong_FromUnsignedLong((unsigned long) tile)
+            ); 
+            cornerIndex++;
+        });
+
+        return list; 
+    }
+
+    PyErr_SetString(PyExc_AttributeError, "for_player must be valid or None"); 
+    return NULL; 
+}
+
+static PyObject* Board_NumPlayerOpenCorners(BoardObject* self, PyObject* args, PyObject* kwds) 
```

Comment:
`board.n_player_corners` in Python; just a count of how many open corners the player has, less work than getting the list and then taking the length when you only care about the count in Python